### ISSUE TITLE
Nested AND + OR returns zero rows

### ISSUE THREAD

**Issue by pomgui** [oguimbal/pg-mem#201]
# Describe the bug

In a query with nested AND/OR where conditions, the query returns zero records

# To Reproduce

```sql
CREATE TABLE test(id serial, a int, b int);
INSERT INTO test(a,b) VALUES (1, 2), (3, 4);

-- Everything is ok with a simple OR condition (both in postgres and pg-mem):
SELECT *
FROM   test
WHERE (a = 1 AND b = 2)
   OR (a = 3 AND b = 4);
 id | a | b 
----+---+---
  1 | 1 | 2
  2 | 3 | 4
(2 rows)

-- The next query only returns records in postgres, not in pg-mem (nor in the playground)
SELECT *
FROM   test
WHERE id > 0 AND (
      (a = 1 AND b = 2)
   OR (a = 3 AND b = 4)
);
 id | a | b 
----+---+---
  1 | 1 | 2
  2 | 3 | 4
(2 rows)

```

# Workaround
It is possible to get the where condition records using subqueries:

```sql
SELECT * FROM (
	SELECT *
	FROM   test
	WHERE (a = 1 AND b = 2)
	   OR (a = 3 AND b = 4)
) x
WHERE id > 0;
 id | a | b 
----+---+---
  1 | 1 | 2
  2 | 3 | 4
(2 rows)
```

# pg-mem version

```
node -p "require('./node_modules/pg-mem/package.json').version"
2.3.3
```

**Comment by oguimbal** [comment]
Wow, that's was nasty one 😣 I'm quite surprised this passed through the unit tests.

Many thanks for this investiguation and for taking the time to report that.

Fixed in `pg-mem@2.3.4`

**Comment by pomgui** [comment]
I'm actively developing various projects with pg, and after discover your project I refactored my tests to use pg-mem instead of dockerized pg instances.
This bug was really a nasty one :-) (it made me run in circles until I find it)
Anyway, it's great having guys like you providing amazing tools like this.
Thank you! 

BTW. The playground still has the bug, I imagine it's using a previous version, perhaps you can inform the version (something like "powered by pg-mem@x.y.z")?
